Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm

```diff
@@ -180,6 +181,17 @@ - (void)dealloc
     [super dealloc];
 }
 
+- (void)setSupportedOrientation:(UIInterfaceOrientationMask)supportedOrientations
+{
+    _supportedOrientations = supportedOrientations;
+    [self setNeedsUpdateOfSupportedInterfaceOrientations];
+}
+
+- (UIInterfaceOrientationMask)supportedInterfaceOrientations
+{
+    return _supportedOrientations;
```

Comment:
Should this call into the superclass? From looking at `UIViewController`'s logic, it seems that in the `UIUserInterfaceIdiomPhone` case, we use `UIInterfaceOrientationMaskAllButUpsideDown` by default.

Reply:
We can return UIInterfaceOrientationMaskAllButUpsideDown on iPhone instead of UIInterfaceOrientationMaskAll. That said, based on local testing returning `UIInterfaceOrientationMaskAll` on iPhone still doesn't allow you to make the orientation upside down.

I don't understand the comment about calling into the base class. Do you mean that _supportedOrientations would be a std::optional (or whatever the ObjC equivalent of that) and we'd call into the base class if std::nullopt (meaning orientation is unlocked)?

Right now, _supportedOrientations is set unconditionally so...

Reply:
I meant, defaulting `_supportedOrientations` to the value of `[super supportedInterfaceOrientations]` rather than `UIInterfaceOrientationMaskAll`, since UIKit may be choose to be more restrictive than `UIInterfaceOrientationMaskAll` in some user interface idioms.

That said, perhaps it doesn't make any practical difference if (in your testing) it doesn't affect iPhone behavior. Maybe there's more enforcement further down the stack (SpringBoard, BackBoard, etc.)